docs/content/bisync.md

```diff
@@ -0,0 +1,1090 @@
+  beneath `testdata`. For example,
+  `go test ./cmd/bisync -case dry-run -remote gdrive: -remote2 local`
+  refers to the test case in `./cmd/bisync/testdata/test_dry_run`.
+- The test working directory is located at `.../workdir` relative to a
+  temporary test directory, usually under `/tmp` on Linux.
+- The local test sync tree is created at a temporary directory named
+  like `bisync.XXX` under system temporary directory.
+- The remote test sync tree is located at a temporary directory
+  under `<remote:>/bisync.XXX/`.
+- `path1` and/or `path2` subdirectories are created in a temporary
+  directory under the respective local or cloud test remote.
+- By default, the Path1 and Path2 test dirs and workdir will be deleted
+  after each test run. The `-no-cleanup` flag disables purging these
+  directories when validating and debugging a given test.
+  These directories will be flushed before running another test,
+  independent of the `-no-cleanup` usage.
+- You will likely want to add `- /testdir/` to your normal
+  bisync `--filters-file` so that normal syncs do not attempt to sync
+  the test temporary directories, which may have `RCLONE_TEST` miscompares
+  in some testcases which would otherwise trip the `--check-access` system.
+  The `--check-access` mechanism is hard-coded to ignore `RCLONE_TEST`
+  files beneath `bisync/testdata`, so the test cases may reside on the
+  synched tree even if there are check file mismatches in the test tree.
+- Some Dropbox tests can fail, notably printing the following message:
+  `src and dst identical but can't set mod time without deleting and re-uploading`
+  This is expected and happens due a way Dropbox handles modificaion times.
+  You should use the `-refresh-times` test flag to make up for this.
+- If Dropbox tests hit request limit for you and print error message
+  `too_many_requests/...: Too many requests or write operations.`
+  then follow the
+  [Dropbox App ID instructions](/dropbox/#get-your-own-dropbox-app-id).
+
+### Updating golden results
+
+Sometimes even a slight change in the bisync source can cause little changes
+spread around many log files. Updating them manually would be a nighmare.
+
+The `-golden` flag will store the `test.log` and `*.lst` listings from each
+test case into respective golden directories. Golden results will
+automatically contain generic strings instead of local or cloud paths which
+means that they should match when run with a different cloud service.
+
+Your normal workflow might be as follows:
+1. Git-clone the rclone sources locally
+2. Modify bisync source and check that it builds
+3. Run the whole test suite `go test ./cmd/bisync -remote local`
+4. If some tests show log difference, recheck them individually, e.g.:
+   `go test ./cmd/bisync -remote local -case basic`
+5. If you are convinced with the difference, goldenize all tests at once:
+   `go test ./cmd/bisync -remote local -golden`
```

Comment:
@ncw @cjnaz

The `rclonetest -golden` was **the thing** that bought me.
It polished it a little with _anti-hoppers_ (a far relative of [sorting in fstest.CompareItems](https://github.com/rclone/rclone/blob/v1.55.1/fstest/fstest.go#L391)) and [replacers](https://github.com/rclone/rclone/pull/5164/files#diff-ee6acf898897fdf3c0a0b890fc8feffdcac69b41323b24da034da40af46df01bR1145) and gave me a mechanism to snapshot a **bigger picture**.
Git's colorful word diff then gives a tool to lead the **continuous refactoring** with all noise filtered away.
I think this approach can have its chance make into the rclone test suite alongside with [fstest framework](https://github.com/rclone/rclone/blob/v1.55.1/fstest/fstest.go) ~~and live happily together ever after~~.